opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -55,5 +54,23 @@
    * (Note that some OpenTracing implementations consider the Spans on either side of an RPC to have the same identity,
    * and others consider the caller to be the parent and the receiver to be the child)
    */
-  <T> Span join(String operationName, T carrier);
+  <T> SpanBuilder join(String operationName, T carrier);
```

Comment:
... oh, but there is a catch: namely that there are circumstances where no operation name is available at `join()` time (hence `Span.setOperationName()`). So an assertion won't work, or at least not that particular assertion. Another possibility would be to add something like

```
  SpanBuilder withoutInitialOperationName();
```

(or similar) which would force the user to specifically opt out of an operation name at `start()`-time.